### PR TITLE
[MO] - [system test] -> TracingST

### PR DESCRIPTION
**Pull request by Seequick1**
### Type of change

- Enhancement / new feature
- Refactoring

### Description

This PR adds the following things. 
- new Test Suite, for the tracing based on `Jaeger`
- new HttpUtils
- new Deployments of `consumer, producer, kafka-streams and kafka-connect`
- new Dependency `rest-assured` to test `Jaeger` more effective

### Checklist

- [x] Write tests
- [x] Make sure all tests pass

### CODE REVIEW

File: systemtest/src/test/java/io/strimzi/systemtest/tracing/TracingST.java [hunk 1]
```diff
@@ -0,0 +1,785 @@
...
+        configOfSourceKafka.put("offsets.topic.replication.factor", "1");
+        configOfSourceKafka.put("transaction.state.log.replication.factor", "1");
+        configOfSourceKafka.put("transaction.state.log.min.isr", "1");
+
+        testMethodResources().kafkaEphemeral(CLUSTER_NAME, 1, 1)
+                .editSpec()
+                    .editKafka()
+                        .withConfig(configOfSourceKafka)
+                        .withNewPersistentClaimStorage()
+                            .withNewSize("100")
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/tracing/TracingST.java]:
```suggestion
                            .withNewSize("10")
```
On some of our environments we have PVs only with 10Gb available

File: systemtest/src/test/java/io/strimzi/systemtest/tracing/TracingST.java [hunk 2]
```diff
@@ -0,0 +1,785 @@
...
+    }
+
+    @Test
+    void testProducerService() {
+        Map<String, Object> configOfSourceKafka = new HashMap<>();
+        configOfSourceKafka.put("offsets.topic.replication.factor", "1");
+        configOfSourceKafka.put("transaction.state.log.replication.factor", "1");
+        configOfSourceKafka.put("transaction.state.log.min.isr", "1");
+
+        testMethodResources().kafkaEphemeral(CLUSTER_NAME, 1, 1)
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/tracing/TracingST.java]: Ephemeral with persistent storage doesn't look good. Don't we have persistent kafka in resources?

File: systemtest/src/test/java/io/strimzi/systemtest/tracing/TracingST.java [hunk 3]
```diff
@@ -0,0 +1,785 @@
...
+        assertThat(serviceName + " service doesn't produce traces", data.size(), greaterThan(0));
+    }
+
+    private void verifyServicesHasSomeTraces(String... serviceNames) {
+        for (String serviceName : serviceNames) {
+            verifyServicesHasSomeTraces(serviceName);
+        }
+    }
+
+    private void installJaeger() {
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/tracing/TracingST.java]:
```suggestion
    private void deployJaeger() {
```

**Comment by Seequick1** [comment]
@strimzi-ci run tests profile=tracing testcase=TracingST

**Comment by Seequick1** [comment]
@strimzi-ci run tests profile=tracing testcase=TracingST#testProducerConsumerMirrorMakerConnectStreamsService

**Comment by Seequick1** [comment]
@strimzi-ci run tests profile=tracing testcase=TracingST#testProducerConsumerMirrorMakerConnectStreamsService

**Comment by Seequick1** [comment]
@strimzi-ci run tests profile=tracing testcase=TracingST#testProducerConsumerMirrorMakerConnectStreamsService

**Pull request merged**